### PR TITLE
[sw, base] Explode the `base:base` rule into many smaller rules

### PR DESCRIPTION
**Pull request by mcy**
As it was before, it's very difficult to appropriately test things inside of base or thread threw delicate flags; it also makes it very hard to tell who's using what pieces of base. This CL resolves both, partially; `base:base` still exists, but is deprecated and is only used inside of `silicon_creator`, which I plan to clean up separately.

### CODE REVIEW

File: sw/device/lib/base/BUILD [hunk 1]
```diff
@@ -7,58 +7,107 @@ package(default_visibility = ["//visibility:public"])
...
+    hdrs = ["abs_mmio.h"],
+    target_compatible_with = [OPENTITAN_CPU],
+    deps = [":mmio"],
+)
+
 # tests need to build src with a different set of flags. use this to do so.
 filegroup(
     name = "mmio_src",
     srcs = ["mmio.c"],
+    deprecation = "Please use :mmio instead, if possible.",
```
Comment (drewmacrae) [inline sw/device/lib/base/BUILD:112]: If we depend on mmio how do we get the version built with MOCK_MMIO for unit tests and the version built without it for software built for OpenTitan?

**Review by drewmacrae — APPROVED** [review]
SGTM, @cfrantz and I discussed this at some point and I'd like him to weigh in because I think he was originally on the side of collecting it to one target, but as I understand this there's now a reason to break it up, it's not just being broken up because meson did it that way and they were the best record I had of intent regarding how to build.

File: sw/device/lib/base/BUILD (hunk 1 above)
Comment (mcy) [reply sw/device/lib/base/BUILD:112]: The MOCK_MMIO version should not depend on `mmio.c`, and provide those symbols itself. I'm going to see if I can just kill this target.

**Comment by drewmacrae** [comment]
https://github.com/lowRISC/opentitan/pull/8147#discussion_r713291419

**Comment by drewmacrae** [comment]
There's a typo in one of the commit messages that probably wont mater when this gets squashed into a single commit

**Review by cfrantz — APPROVED** [review]

**Review by alphan — APPROVED** [review]

**Pull request merged**